From 469319d402ab844778cf7fbcd94ba1c9d9cfc9ed Mon Sep 17 00:00:00 2001
From: Leif Lindholm <leif.lindholm@linaro.org>
Date: Thu, 27 Nov 2014 17:56:07 +0000
Subject: [PATCH 2/2] of: support passing console options with stdout-path
Origin: https://git.kernel.org/cgit/linux/kernel/git/glikely/linux.git/commit/?id=7914a7c5651a51617d952e8fa745000ed8c4f001

Support specifying console options (like with console=ttyXN,<options>)
by appending them to the stdout-path property after a separating ':'.

Example:
        stdout-path = "uart0:115200";

Signed-off-by: Leif Lindholm <leif.lindholm@linaro.org>
[grant.likely: minor rework to shorten the diffstat]
Signed-off-by: Grant Likely <grant.likely@linaro.org>
---
 drivers/of/base.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/of/base.c b/drivers/of/base.c
index da9a8d2..bad67ca 100644
--- a/drivers/of/base.c
+++ b/drivers/of/base.c
@@ -37,6 +37,7 @@ EXPORT_SYMBOL(of_allnodes);
 struct device_node *of_chosen;
 struct device_node *of_aliases;
 static struct device_node *of_stdout;
+static const char *of_stdout_options;
 
 static struct kset *of_kset;
 
@@ -2079,7 +2080,7 @@ void of_alias_scan(void * (*dt_alloc)(u64 size, u64 align))
 		if (!name)
 			name = of_get_property(of_chosen, "linux,stdout-path", NULL);
 		if (name)
-			of_stdout = of_find_node_by_path(name);
+			of_stdout = of_find_node_opts_by_path(name, &of_stdout_options);
 	}
 
 	of_aliases = of_find_node_by_path("/aliases");
@@ -2206,7 +2207,8 @@ bool of_console_check(struct device_node *dn, char *name, int index)
 {
 	if (!dn || dn != of_stdout || console_set_on_cmdline)
 		return false;
-	return !add_preferred_console(name, index, NULL);
+	return !add_preferred_console(name, index,
+				      kstrdup(of_stdout_options, GFP_KERNEL));
 }
 EXPORT_SYMBOL_GPL(of_console_check);
 
-- 
2.1.3

